### PR TITLE
feat(processing_time_checker): add a new package

### PR DESCRIPTION
**Pull request by takayuki5168**
## Description

launcher PR: https://github.com/autowarefoundation/autoware_launch/pull/1072

Added a processing_time_checker to publish diagnotics for the processing time using the input of the processing time of each module.

The default config tracks the following modules.
```
[processing_time_checker_node-1] lateral
[processing_time_checker_node-1] longitudinal
[processing_time_checker_node-1] lane_departure_checker_node
[processing_time_checker_node-1] map_based_prediction
[processing_time_checker_node-1] multi_object_tracker
[processing_time_checker_node-1] planning_validator
[processing_time_checker_node-1] avoidance_by_lane_change
[processing_time_checker_node-1] dynamic_obstacle_avoidance
[processing_time_checker_node-1] goal_planner
[processing_time_checker_node-1] lane_change_left
[processing_time_checker_node-1] lane_change_right
[processing_time_checker_node-1] side_shift
[processing_time_checker_node-1] start_planner
[processing_time_checker_node-1] static_obstacle_avoidance
[processing_time_checker_node-1] behavior_path_planner
[processing_time_checker_node-1] blind_spot
[processing_time_checker_node-1] crosswalk
[processing_time_checker_node-1] detection_area
[processing_time_checker_node-1] intersection
[processing_time_checker_node-1] merge_from_private
[processing_time_checker_node-1] no_stopping_area
[processing_time_checker_node-1] run_out
[processing_time_checker_node-1] stop_line
[processing_time_checker_node-1] walkway
[processing_time_checker_node-1] elastic_band_smoother
[processing_time_checker_node-1] dynamic_obstacle_stop
[processing_time_checker_node-1] obstacle_velocity_limiter
[processing_time_checker_node-1] out_of_lane
[processing_time_checker_node-1] motion_velocity_planner
[processing_time_checker_node-1] obstacle_cruise_planner
[processing_time_checker_node-1] path_optimizer
[processing_time_checker_node-1] velocity_smoother
[processing_time_checker_node-1] shape_estimation
```

```
$ ros2 topic echo /system/processing_time_checker/metrics
header:
  stamp:
    sec: 1720677749
    nanosec: 471124532
  frame_id: ''
status:
- level: "\0"
  name: processing_time
  message: ''
  hardware_id: ''
  values:
  - key: lane_departure_checker_node
    value: '1.072000'
  - key: motion_velocity_planner
    value: '17.773000'
  - key: dynamic_obstacle_avoidance
    value: '0.123000'
  - key: goal_planner
    value: '0.718000'
  - key: intersection
    value: '9.437000'
  - key: lane_change_left
    value: '0.065000'
  - key: behavior_path_planner
    value: '6.986000'
  - key: multi_object_tracker
    value: '0.066631'
  - key: side_shift
    value: '0.808000'
```
## Related links

**Parent Issue:**

- Link

## How was this PR tested?

The output from the node can be connected to the diagnostic_converter as follows.
```
$ ros2 node info /diagnostic_converter
...
    /system/processing_time_checker/metrics_processing_time_avoidance_by_lane_change: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_behavior_path_planner: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_blind_spot: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_crosswalk: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_detection_area: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_dynamic_obstacle_avoidance: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_dynamic_obstacle_stop: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_elastic_band_smoother: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_goal_planner: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_intersection: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_lane_change_left: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_lane_change_right: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_lane_departure_checker_node: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_lateral: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_longitudinal: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_map_based_prediction: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_merge_from_private: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_motion_velocity_planner: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_multi_object_tracker: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_no_stopping_area: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_obstacle_velocity_limiter: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_out_of_lane: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_planning_validator: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_run_out: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_shape_estimation: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_side_shift: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_start_planner: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_static_obstacle_avoidance: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_stop_line: tier4_simulation_msgs/msg/UserDefinedValue
    /system/processing_time_checker/metrics_processing_time_walkway: tier4_simulation_msgs/msg/UserDefinedValue
```
## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

**Review by kosuke55 — APPROVED** [review]
LGTM!

File: system/autoware_processing_time_checker/src/processing_time_checker.hpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 TIER IV, Inc.
```
Comment (kosuke55) [inline system/autoware_processing_time_checker/src/processing_time_checker.hpp:1]:
```suggestion
// Copyright 2024 TIER IV, Inc.
```

File: system/autoware_processing_time_checker/src/processing_time_checker.cpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 TIER IV, Inc.
```
Comment (kosuke55) [inline system/autoware_processing_time_checker/src/processing_time_checker.cpp:1]:
```suggestion
// Copyright 2024 TIER IV, Inc.
```

File: system/autoware_processing_time_checker/src/processing_time_checker.hpp (hunk 1 above)
Comment (takayuki5168) [reply system/autoware_processing_time_checker/src/processing_time_checker.hpp]: Thank you. Fixed.

File: system/autoware_processing_time_checker/src/processing_time_checker.cpp (hunk 2 above)
Comment (takayuki5168) [reply system/autoware_processing_time_checker/src/processing_time_checker.cpp]: Thank you. Fixed.

**Pull request merged**